File: rpg-frontend/src/types/battle.ts
export interface BattleLogEntryOut {
  turn: number;
  attacker: string;
  target: string;
  action_type: string;
  skill_id: number | null;
  damage: number;
  critical: boolean;
  target_hp_after: number;
  status_effects: Record<string, unknown>[];
}

export interface EnemySummaryOut {
  id: number;
  name: string;
  level: number;
}

export interface BattleOut {
  id: number;
  enemy: EnemySummaryOut;
  result: "won" | "lost";
  turns: number;
  log: BattleLogEntryOut[];
  reward_xp: number;
  reward_coins: number;
  hero_level: number;
  hero_xp: number;
  balance: number;
  created_at: string;
}

export interface StartBattleRequest {
  enemy_template_id: number;
  idempotency_key?: string | null;
}

File: rpg-backend/app/schemas/battle.py
from pydantic import BaseModel


class BattleLogEntryOut(BaseModel):
    turn: int
    attacker: str
    target: str
    action_type: str
    skill_id: int | None
    damage: int
    critical: bool
    target_hp_after: int
    status_effects: list[dict]


class EnemySummaryOut(BaseModel):
    id: int
    name: str
    level: int


class BattleOut(BaseModel):
    id: int
    enemy: EnemySummaryOut
    result: str
    turns: int
    log: list[BattleLogEntryOut]
    reward_xp: int
    reward_coins: int
    hero_level: int
    hero_xp: int
    balance: int
    created_at: str


class StartBattleRequest(BaseModel):
    enemy_template_id: int
    idempotency_key: str | None = None
